[bug] wrong dependency declaration

### Problem

The `abortable-iterator` package is a direct dependency as it is required by `src/dialer/index.js`. But it is declared in the `devDependencies` section which results in a missing module error.

### Solution

move `abortable-iterator` from devDeps to deps

bump

`D:/React-WorkSpace/COSC2410_SEM_2_2021/.yarn/cache/libp2p-npm-0.32.4-fe607aff53-99957c9e00.zip/node_modules/libp2p/src/dialer/index.js
Module not found: libp2p tried to access abortable-iterator, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
`

--- a/package.json
+++ b/package.json
@@ -78,6 +78,7 @@
     ]
   },
   "dependencies": {
+    "abortable-iterator": "^3.0.0",
     "@motrix/nat-api": "^0.3.1",
     "@vascosantos/moving-average": "^1.1.0",
     "abort-controller": "^3.0.0",
@@ -136,7 +137,6 @@
     "@types/node": "^16.0.1",
     "@types/node-forge": "^0.10.1",
     "@types/varint": "^6.0.0",
-    "abortable-iterator": "^3.0.0",
     "aegir": "^33.1.1",
     "buffer": "^6.0.3",
     "delay": "^5.0.0",

